mm2src/mm2_main/src/rpc/lp_commands/eth.rs

```diff
@@ -0,0 +1,84 @@
+//! This source file is for RPCs specific for EVM platform
+
+use coins::eth::{u256_to_big_decimal, wei_from_big_decimal, EthCoin, Web3RpcError};
+use coins::{lp_coinfind_or_err, MmCoin, MmCoinEnum, NumConversError, Transaction, TransactionErr};
+use common::HttpStatusCode;
+use derive_more::Display;
+use enum_derives::EnumFromStringify;
+use ethereum_types::Address as EthAddress;
+use futures::compat::Future01CompatExt;
+use http::StatusCode;
+use mm2_core::mm_ctx::MmArc;
+use mm2_err_handle::{mm_error::MmError, prelude::MmResult};
+use mm2_number::BigDecimal;
+use rpc::v1::types::Bytes as BytesJson;
+
+#[derive(Debug, Deserialize)]
+pub struct Erc20ApproveRequest {
+    coin: String,
+    spender: EthAddress,
+    amount: BigDecimal,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Erc20AllowanceRequest {
+    coin: String,
+    spender: EthAddress,
+}
```

Comment:
It's better to group related requests/responses and methods with each other instead of grouping requests for different methods together.

Reply:
changed